nomad/structs/structs.go

```diff
@@ -5350,6 +5350,34 @@ func (t *Task) Validate(ephemeralDisk *EphemeralDisk, jobType string) error {
 		}
 	}
 
+	// Validation for Kind field which is used for Consul Connect integration
+	// TODO better wording for all error messages
+	if strings.HasPrefix(t.Kind, "connect:") {
+		// This task is a Connect proxy so it should not have service stanzas
+		if len(t.Services) > 0 {
+			mErr.Errors = append(mErr.Errors, fmt.Errorf("Connect proxy task should not have service stanza in it"))
+		}
+		if t.Leader {
+			mErr.Errors = append(mErr.Errors, fmt.Errorf("Connect proxy task must not have leader set"))
+		}
+
+		parts := strings.Split(t.Kind, ":")
+		serviceName := strings.Join(parts[1:], "")
+		if len(parts) > 2 {
+			mErr.Errors = append(mErr.Errors, fmt.Errorf("Proxy service name %q should not contain `:`", serviceName))
```

Comment:
Let's be consistent and always call it "Connect proxy ..." in errors and logs.

```suggestion
			mErr.Errors = append(mErr.Errors, fmt.Errorf("Connect proxy service name %q should not contain `:`", serviceName))
```